tests/benchmark/benchmark_test.go

```diff
@@ -0,0 +1,252 @@
+// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package benchmark
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+)
+
+func BenchmarkApi(b *testing.B) {
+	err, client, ctx := createGrpcClient(b)
+	if err != nil {
+		return
+	}
+	operations := [5]string{"Create", "Update", "Get", "List_pagesize_20", "Delete"}
+
+	for _, operation := range operations {
+		b.Run(fmt.Sprintf("%s", operation), func(b *testing.B) {
+			for i := 1; i <= b.N; i++ {
+				apiId := getApiName(i)
+				switch operation {
```

Comment:
This is an interesting structure (almost like a state machine?) which I think might be intended to provide some flexibility that isn't being used yet (since `operations` above is a fixed array). If we wanted a way to repeatedly do a series of things in a standard wrapper, I'd suggest having an array of Go [anonymous functions](https://gobyexample.com/closures) instead, but before doing that I would probably just write one instance in a simpler, possibly more tedious way and abstract from there.